File: scratch/count_braces.py

with open('src/app/features/courses/courses.component.html', 'r', encoding='utf-8') as f:
    content = f.read()
    open_count = content.count('{')
    close_count = content.count('}')
    print(f"Open: {open_count}, Close: {close_count}")
    
    # Check for unclosed {{
    double_open = content.count('{{')
    double_close = content.count('}}')
    print(f"Double Open: {double_open}, Double Close: {double_close}")
    
    # Check for @if/{ pairs
    at_if = content.count('@if')
    at_for = content.count('@for')
    print(f"@if: {at_if}, @for: {at_for}")

File: scratch/count_braces.js

const fs = require('fs');
const content = fs.readFileSync('src/app/features/courses/courses.component.html', 'utf8');
const openCount = (content.match(/{/g) || []).length;
const closeCount = (content.match(/}/g) || []).length;
console.log(`Open: ${openCount}, Close: ${closeCount}`);

const doubleOpen = (content.match(/{{/g) || []).length;
const doubleClose = (content.match(/}}/g) || []).length;
console.log(`Double Open: ${doubleOpen}, Double Close: ${doubleClose}`);

const controlFlowOpen = (content.match(/@\w+\s*\(.*?\)\s*{/g) || []).length;
console.log(`Control Flow Open: ${controlFlowOpen}`);
